### PR TITLE
LPS-130869

### PR DESCRIPTION
**Pull request by martamedio**
Hey @topolik

I have detected that your change it's not enough to fix the token introspection action, this is a regression bug after having made changes to some constants during _Remember me/Trusted App_  OAuth2 features.

We were [asked to standardize the values of the constants](https://github.com/brianchandotcom/liferay-portal/pull/100022#issuecomment-801941795) so that the separation was a point, but we did not update the [feature value in the Portlet](https://github.com/liferay-appsec/liferay-portal/commit/d255a1b44178700327707a8ce9a4f4cc6b0b1f2c); I have added the solution and an integration test.

@stian-sigvartsen please review it before forward

Thanks!

### CODE REVIEW

File: modules/apps/oauth2-provider/oauth2-provider-web/src/main/resources/content/Language.properties [hunk 1]
```diff
@@ -103,7 +103,7 @@ the-client-will-impersonate-the-selected-client-credential-user-but-will-be-rest
 this-application-wants-the-following-permissions=This application wants the following permissions:
```
Comment (liferay-appsec) [inline modules/apps/oauth2-provider/oauth2-provider-web/src/main/resources/content/Language.properties:103]: Check that added properties are either titles or sentences

**Review by liferay-appsec — COMMENTED** [review]
This is an automatic review suggestion

**Comment by martamedio** [comment]
ci:test:sf

**Comment by martamedio** [comment]
ci:test:relevant

**Comment by topolik** [comment]
Hi @martamedio,

thanks for checking.

**Comment by stian-sigvartsen** [comment]
@martamedio  As we just discussed. The change to the default-portlet-preferences.xml is introducing a regression because the OAuth 2 portlet's default preferences are already stored into DB and will not get recreated automatically. So that would need an upgrade process. It looks like it "works" because we're also missing an upgrade process for existing OAuth2Applications that store the portlet preference value in its "features" column.

So we could get away with the current situation, however if the stored portlet preferences somehow get reset, thing no longer match.

**Comment by martamedio** [comment]
ci:test:sf

**Comment by martamedio** [comment]
ci:test:relevant

**Comment by martamedio** [comment]
ci:test:relevant

**Comment by martamedio** [comment]
ci:test:relevant

**Comment by stian-sigvartsen** [comment]
Hey @martamedio . I tweaked the way the upgrade process identifies features to avoid accidents like "something_token_introspection" being corrupted by the upgrade. Highly unlikely of course, but if it ever happened it would be difficult to rectify.

Also, I think we are in a strict sense actually introducing both a major schema version bump AND a major package version bump. But I wasn't able to get "gw baseline" to recognize the latter. Thoughts?

**Comment by martamedio** [comment]
Hey @stian-sigvartsen 

I disagree with [this change](https://github.com/liferay-appsec/liferay-portal/pull/341/commits/daba353d2d3296358392f9b5330ed6f1fb08be13): if a client has more Portlet Features implemented and is also using the existing "_token_introspection_" feature, with your change our string will not be replaced.
I see it more likely that there are extra features implemented with a different name than someone using a value like "_something_token_introspection_".

About the bump: considering that adding a column to a table is a minor change, changing the value of it I think it would be as well, but see what @lipusz thinks.

**Comment by martamedio** [comment]
@stian-sigvartsen Edited my previous comment: I misunderstood the logic, it does correctly cover the two cases referred to, sorry! 😅

**Comment by stian-sigvartsen** [comment]
@martamedio . Strictly speaking, minor SemVer of schema changes requires the new schema to work with old code.  In this case it wouldn't work because the old code would be setting/comparing with "token_instrospection". So I think we should major bump the schema. But as discussed offline, we don't need to major bump the packages

**Comment by martamedio** [comment]
Please, see: https://github.com/liferay-appsec/liferay-portal/pull/353 and https://github.com/liferay-appsec/liferay-portal/pull/352

**Pull request closed without merging**